### PR TITLE
Add tools to convert Pinot segment into another format.

### PR DESCRIPTION
**Pull request by Jackie-Jiang**
1. Fix bug in PinotSegmentRecordReader when reading multi-value columns.
2. Add multi-value support to PinotSegmentToAvroConverter.
3. Add CSV/JSON converter.

### CODE REVIEW

File: pinot-tools/src/main/java/com/linkedin/pinot/tools/segment/converter/PinotSegmentConvertCommand.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+@SuppressWarnings("FieldCanBeLocal")
+public class PinotSegmentConvertCommand extends AbstractBaseCommand implements Command {
+
+  @Option(name = "-segmentDir", required = true, metaVar = "<String>", usage = "Path to Pinot segment directory.")
+  private String _segmentDir;
+
+  @Option(name = "-outputFile", required = true, metaVar = "<String>", usage = "Path to output file.")
+  private String _outputFile;
+
+  @Option(name = "-fileFormat", required = true, metaVar = "<String>", usage = "Format to convert to (AVRO/CSV/JSON).")
```
Comment (mayankshriv) [inline pinot-tools/src/main/java/com/linkedin/pinot/tools/segment/converter/PinotSegmentConvertCommand.java]: Rename -fileFormat to -outputFormat

File: pinot-tools/src/main/java/com/linkedin/pinot/tools/segment/converter/PinotSegmentConvertCommand.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  @Option(name = "-overwrite", required = false, metaVar = "<boolean>",
+      usage = "Overwrite the existing file (default false).")
+  private boolean _overwrite;
+
+  @Option(name = "-help", required = false, help = true, aliases = {"-h", "--h", "--help"},
+      usage = "Print this message.")
+  private boolean _help;
+
+  @Override
+  public boolean execute()
```
Comment (mayankshriv) [inline pinot-tools/src/main/java/com/linkedin/pinot/tools/segment/converter/PinotSegmentConvertCommand.java]: Can this be enhanced to handle the following:
1. Segments in tar.gz format.
2. Case when segmentDir contains a list of segments, and not just one segment.

Comment (Jackie-Jiang) [inline pinot-tools/src/main/java/com/linkedin/pinot/tools/segment/converter/PinotSegmentConvertCommand.java]: We can do that, but in my opinion, it's not necessary. What will the output file name be if there are several segments? If we want to do batch conversion, all we need is a wrapper and then convert them one by one.

File: pinot-tools/src/main/java/com/linkedin/pinot/tools/segment/converter/PinotSegmentConvertCommand.java (hunk 1 above)
Comment (Jackie-Jiang) [inline pinot-tools/src/main/java/com/linkedin/pinot/tools/segment/converter/PinotSegmentConvertCommand.java]: Fixed.

**Pull request merged**